framework/decode/json_writer.cpp

```diff
@@ -204,9 +204,9 @@ std::string JsonWriter::GenerateFilename(const std::string_view filename)
 bool JsonWriter::WriteBinaryFile(const std::string& filename, uint64_t data_size, const uint8_t* data)
 {
     FILE* file_output = nullptr;
-    if (util::platform::FileOpen(&file_output, filename.c_str(), "wb") == 0)
+    if (util::platform::FileOpen(&file_output, filename.c_str(), "wb") == 0 &&
+        util::platform::FileWrite(data, static_cast<size_t>(data_size), file_output))
```

Comment:
Aren't there scenarios where this leave open the file if the write fails for some error other than the ones you expect?

Reply:
Latest changes seem to fix this.